### ISSUE TITLE
Close tab button is too big

### ISSUE THREAD

**Issue by jamesonquinn** [atom/atom#11448]
### Description

It's way too easy to close tabs by mistake. This is especially annoying for the julia console tab, which cannot be reopened in the same place with built-in shortcuts. 

### Steps to Reproduce

1. Open a bunch of tabs
2. Click on a different tab to select it
3. Whoops, closed it by mistake
4. Grit teeth, reopen

**Expected behavior:** [What you expected to happen]

In google chrome, only the currently active tab has a close button. To close a different tab takes two clicks; one to select, one to close.

**Actual behavior:** [What actually happened]

The close button (box? what do you call it?) actually has an even bigger active area than it looks like, so anywhere towards the right side of the text ends up closing the tab. Ideally, you'd reproduce the Chrome-like behavior as above; but even if not, the close button cannot be sensitive to clicks over title text, so make the active area smaller please.

### Versions

I'm on 1.6.2, but probably all versions.

(Note: Atom is awesome in general, but this makes me angry every time.)

**Comment by mnquintana** [comment]
What theme are you using?

**Comment by damieng** [comment]
Also, Chrome has a close button on every tab. See screenshot
![image](https://cloud.githubusercontent.com/assets/118951/14506499/f28256ce-0172-11e6-9860-725f9465d638.png)

**Comment by jacobsmith** [comment]
This may not fix the issue, but I find myself using `Cmd + Shift + t` on my mac quite often to reopen the last closed tab. That may be a useful workaround for now.

**Comment by jamesonquinn** [comment]
1. my theme is "one dark", which is what it installed as
2. Yes, Chrome has close buttons on every tab when the tabs are big, but when they get many and small, the close buttons disappear except on the active one.
3. cmd-shift-t doesn't work to reopen the julia console.

**Comment by JustRiedy** [comment]
![000048](https://cloud.githubusercontent.com/assets/14937219/14517947/81e1ad24-0245-11e6-9fca-7334152aab68.png)
Here's an image to illustrate @jamesonquinn 's issue.
The tab I have my mouse over isn't active, but it still has a close button that takes up a third of its UI. 

Very easy to accidentally click on when trying to open the tab.

**Comment by Ben3eeE** [comment]
I never close tabs with this button so I also found it intrusive. I use Mouse3 or ctrl-w for all tabs I close.

There is a package https://atom.io/packages/atom-no-tab-close-button that removes the close button from all tabs, I have been using this since the day I installed atom. Perhaps this package can be modified to get the behavior of showing the button only for the current active tab if you really want that.

**Comment by gavsiu** [comment]
@Ben3eeE, looking at that package, it seems that it removes the close-icon completely, which also doubles as the modified icon. I tried copying the styles without installing the package and it seems to confirm my suspicions.

I want the modified icon and also the close button for the current tab only. What I've come up with isn't perfect, but it's the best I could do without modifying the tab package.

    .tab-bar .tab:hover:not(.active) .close-icon {
    	display: none;
    }

    .tab-bar .tab.modified:not(:hover) .close-icon {
    	display: block;
    }

Modified icons shown by default. Close icon only visible on hover in active tab. Clicking in the close icon area on other tabs only changes to that tab because the close button is hidden.

The downside is modified icons are also hidden on hover if the tab is not active. The close action is probably a script command tied to div.close-icon, so I can't really prevent the close action with CSS. Not much I can do without modifying more files other than the stylesheet.

Let me know if anyone comes up with a better solution.

**Comment by jamesonquinn** [comment]
gavsiu: what file should I put that css into?

2016-05-28 0:34 GMT-04:00 gavsiu <notifications@github.com>:

**Comment by gavsiu** [comment]
Atom menu > Edit > Styleshet
Save
Atom menu > View > Developer > Reload Window

**Comment by Ben3eeE** [comment]
@gavsiu Nice I never realized that I was missing out on the modified icon.

What about using css to change the size of the clickable area? Instead of removing it completely or trying to hide and unhide it based on various states. Or I guess you could change the size in addition to your hiding and unhiding for maximum protection against misclicks.

I came up with the following that works with `one-dark` for me, maybe someone that understands css can improve on this and make it work with more themes.

Two problems, horizontally the button is positioned with padding that is clickable, so let's clear this and position the button with margin. Vertically the size of the button is set to be the full height so we change height and line-height to be just the size we want and then position it with margin.

```
.tab-bar .tab .close-icon
{
    padding: 0em;
    margin-right: 0.8em;
    margin-top: 10px;
    line-height: 12px;
    height: 12px;
}
```

**Comment by chmercesmoreira** [comment]
I posted a duplicated accidentally here: https://github.com/atom/tabs/issues/348
My sample image of this "issue": 
![atom](https://cloud.githubusercontent.com/assets/20016536/16172793/1f6cb400-3566-11e6-9b99-cf26ac461c51.png)
When I try to switch to the "gad.h" tab, I almost always end up clicking in the close "x" button. What I suggested was to include a minimum fixed tab size to override the size of those tabs that have a short size due to the name of the file being to short in length.

**Comment by gavsiu** [comment]
@chmercesmoreira If you are using One Dark UI theme, go to theme settings > Tab Sizing > Even

**Comment by chmercesmoreira** [comment]
@gavsiu Thank you, it worked.

**Comment by simurai** [comment]
In Atom `1.10`, the close icon in the One Dark UI theme in minimum mode will have less padding on the sides, making selecting easier for very short file names:

Before | After
--- | ---
<img width="90" alt="screen shot 2016-06-21 at 11 28 27 am" src="https://cloud.githubusercontent.com/assets/378023/16216247/3635e504-37a4-11e6-821b-6bf0c644d5a9.png"> | <img width="96" alt="screen shot 2016-06-21 at 11 27 41 am" src="https://cloud.githubusercontent.com/assets/378023/16216254/4569a592-37a4-11e6-8f89-80eef4cc13f8.png">

> The red is just to show the size

For now, you can use this in your `styles.less`:

```css
.theme-one-dark-ui .tab .close-icon {
  padding-left: .4em;
  padding-right: .4em;
}
```

Or even smaller in @Ben3eeE's example: https://github.com/atom/atom/issues/11448#issuecomment-222436697

Although, making the click area too small, makes it harder to close tabs and you have to aim more precisely. Another option might be to have a background color for the icon on hover, so it's easier to see click area?

**Comment by Ben3eeE** [comment]
@simurai I think having the clickable area smaller as in my example more closely follows how modern browsers work, and I don't think browser tabs are hard to close. I think a behavior/size like Chrome or Opera would be best, where the clickable area is shown on hover. Maybe the size of the X in Atom should be increased just a bit if you decide to use a behavior like this.
As it is now I have a hard time distinguishing between the colors(hover and normal) of Atoms close icon. I can't tell if I am about to close or not just given a color.

Chrome:
![chrome](https://cloud.githubusercontent.com/assets/1058982/16218483/8b22100e-377e-11e6-9e80-f9bb838d480c.gif)
Edge:
![edge](https://cloud.githubusercontent.com/assets/1058982/16218486/948edd66-377e-11e6-9561-3943eba0f7a5.gif)
Opera:
![opera](https://cloud.githubusercontent.com/assets/1058982/16218487/986c266e-377e-11e6-9557-1f6ef874f2ad.gif)
Atom:
![atom](https://cloud.githubusercontent.com/assets/1058982/16218532/1f20e9ce-377f-11e6-8346-0fc0b88faf5f.gif)
Atom with my style:
![atom with benesheet](https://cloud.githubusercontent.com/assets/1058982/16218647/2c67edac-3780-11e6-8512-003257aaf86b.gif)

Just my thoughts :thought_balloon:

**Comment by simurai** [comment]
> I think a behavior/size like Chrome or Opera would be best, where the clickable area is shown on hover.

Yeah, I like that too. I'll try that out for the One themes. 👍

**Comment by simurai** [comment]
There is now a [PR](https://github.com/atom/one-dark-ui/issues/144) that looks like this:

(https://cloud.githubusercontent.com/assets/378023/16225420/5ec9d42a-37e1-11e6-8111-c1d6859b7cc1.gif)

**Comment by lee-dohm** [comment]
Since this has been addressed in the One themes, I'm going to close it. The fix should be available in v1.10 as @simurai mentioned.

Thanks everyone for your help!
